### PR TITLE
feat(rafbila-styrkir): New application

### PR DESCRIPTION
**Pull request by berglindoma13**
# ...

Attach a link to issue if relevant

## What

New application for electric car subsidies

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/templates/energy-funds/src/fields/SelectVehicle/VehicleChekboxField.tsx
Comment (berglindoma13) [reply libs/application/templates/energy-funds/src/fields/SelectVehicle/VehicleChekboxField.tsx:101]: No that's on purpose, there is no radio button component that has this look, would have to be created specifically for this

**Review by AndesKrrrrrrrrrrr — APPROVED** [review]
Infra LGTM

**Review by thordurhhh — APPROVED** [review]
code owned files lgtm

File: libs/clients/energy-funds/src/lib/apiConfiguration.ts
Comment (saevarma) [inline libs/clients/energy-funds/src/lib/apiConfiguration.ts:24]: We should set this to `tokenExchange`

File: libs/clients/energy-funds/src/lib/energyFundsClient.config.ts
Comment (saevarma) [inline libs/clients/energy-funds/src/lib/energyFundsClient.config.ts:21]: Please remove this now, FJS should have stopped using this for new services.

cc @jonarnarbriem

File: libs/clients/energy-funds/src/lib/energyFundsClient.service.ts
Comment (saevarma) [inline libs/clients/energy-funds/src/lib/energyFundsClient.service.ts:19]: `applicationApi` ?

**Review by saevarma — COMMENTED** [review]
Reviewed Core files and have few comments

File: infra/src/dsl/xroad.ts
Comment (berglindoma13) [reply infra/src/dsl/xroad.ts:543]: this is the xroad service registry, I will let them know of the typo and fix it when they do

File: libs/application/templates/energy-funds/src/fields/SelectVehicle/VehicleSelectField.tsx
Comment (Thorrik58) [inline libs/application/templates/energy-funds/src/fields/SelectVehicle/VehicleSelectField.tsx:23]: I would like to revisit the buildSelect fields once this has been released just to see what we are missing in the base field to make this case work.

**Review by Thorrik58 — COMMENTED** [review]
Looks good to me, I'll let Óli have the final review for Norda.
Id like to look into how we can make the buildSelectField function better to suit your case after the release since this needs to go out before end of year.

File: libs/application/template-api-modules/src/lib/modules/templates/energy-funds/energy-funds.service.ts
Comment (obmagnusson) [inline libs/application/template-api-modules/src/lib/modules/templates/energy-funds/energy-funds.service.ts:132]: This is a thrown validation error but the user has no recourse as it has no further information on how to react to the error as it will only read "Umsókn mistókst".  Should at least provide the user with some direction on how to proceed. 
This would be when the user has not selected any vehicles ?

**Review by obmagnusson — COMMENTED** [review]
Just a minor comment from me.

File: libs/application/template-api-modules/src/lib/modules/templates/energy-funds/energy-funds.service.ts
Comment (berglindoma13) [reply libs/application/template-api-modules/src/lib/modules/templates/energy-funds/energy-funds.service.ts]: I realize this case couldn't ever happen as the user would get stuck in the dataProvider stage if they have no vehicles that match the application requirements. I removed this check and just added a null check on the variables in the answer object

File: libs/api/domains/energy-funds/src/lib/energyFunds.service.ts
Comment (saevarma) [inline libs/api/domains/energy-funds/src/lib/energyFunds.service.ts:22]: ~ Looks like a unnecessary wrapping as this private function is only used once. So you should just  use `energyFundsClientService` directly.

File: libs/api/domains/energy-funds/src/lib/graphql/main.resolver.ts
Comment (saevarma) [inline libs/api/domains/energy-funds/src/lib/graphql/main.resolver.ts:32]: ~ `return await` is unnecessary and you should just return the promise.

**Review by saevarma — APPROVED** [review]
LGTM

**Pull request merged**
